docs/source/java/cdata.rst

```diff
@@ -0,0 +1,213 @@
+                <version>${arrow.version}</version>
+            </dependency>
+        </dependencies>
+    </project>
+
+.. code-block:: java
+
+    import org.bytedeco.javacpp.annotation.Platform;
+    import org.bytedeco.javacpp.annotation.Properties;
+    import org.bytedeco.javacpp.tools.InfoMap;
+    import org.bytedeco.javacpp.tools.InfoMapper;
+
+    @Properties(
+            target = "CDataJavaToCppExample",
+            value = @Platform(
+                    include = {
+                            "CDataCppBridge.h"
+                    },
+                    compiler = {"cpp11"},
+                    linkpath = {"/arrow/cpp/build/debug/"},
+                    link = {"arrow"}
+            )
+    )
+    public class CDataJavaConfig implements InfoMapper {
+
+        @Override
+        public void map(InfoMap infoMap) {
+        }
+    }
+
+.. code-block:: shell
+
+    // Compile our Java code
+    javac -cp javacpp-1.5.7.jar CDataJavaConfig.java
+
+    // Generate CDataInterfaceLibrary
+    java -jar javacpp-1.5.7.jar CDataJavaConfig.java
+
+    // Generate libjniCDataInterfaceLibrary.dylib
+    java -jar javacpp-1.5.7.jar CDataJavaToCppExample.java
+
+    // Validate libjniCDataInterfaceLibrary.dylib created
+    otool -L macosx-x86_64/libjniCDataJavaToCppExample.dylib
+    macosx-x86_64/libjniCDataJavaToCppExample.dylib:
+        libjniCDataJavaToCppExample.dylib (compatibility version 0.0.0, current version 0.0.0)
+        @rpath/libarrow.800.dylib (compatibility version 800.0.0, current version 800.0.0)
+        /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1200.3.0)
+        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.0.0)
+
+**Java Test**
+
+Let's create a Java class to Test C Data Interface from Java to C++:
+
+.. code-block:: java
+
+    import org.apache.arrow.c.ArrowArray;
+    import org.apache.arrow.c.ArrowSchema;
+    import org.apache.arrow.c.Data;
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.vector.BigIntVector;
+
+    public class TestCDataInterface {
+        public static void main(String[] args) {
+            try(
+                BufferAllocator allocator = new RootAllocator();
+                ArrowSchema arrowSchema = ArrowSchema.allocateNew(allocator);
+                ArrowArray arrowArray = ArrowArray.allocateNew(allocator)
+            ){
+                CDataJavaToCppExample.fillCArray(
+                        arrowSchema.memoryAddress(), arrowArray.memoryAddress());
+                try(
+                    BigIntVector bigIntVector = (BigIntVector) Data.importVector(
+                            allocator, arrowArray, arrowSchema, null)
+                ){
+                    System.out.println("Java using C Data Interface to read Array Filled by C++: "
+                            + bigIntVector);
+                }
```

Comment:
We should call `release` on the array and schema right?

Reply:
Or hmm. Does closing the vector also close the array for you?

Reply:
It is doing at the end of [ImportVector](https://github.com/apache/arrow/blob/e379ee1f5a3c4de2defbe631f2cbf5567e574f1d/java/c/src/main/java/org/apache/arrow/c/ArrayImporter.java#L63) by c data library

Reply:
That doesn't actually call release. So we do need to call release here.